src/pybind/mgr/dashboard/frontend/src/app/shared/directives/tab-activate-by-fragment.directive.ts

```diff
@@ -0,0 +1,35 @@
+import { Directive, Host, HostListener, Input, OnInit, Optional } from '@angular/core';
+import { Router, UrlTree } from '@angular/router';
+
+import * as _ from 'lodash';
+import { TabDirective } from 'ngx-bootstrap/tabs';
+
+/**
+ * This directive will activate the tab if the route fragment is
+ * identical to its identifier. It will also navigate to the current
+ * route when the tab is selected by adding the tab identifier as
+ * fragment to the current route, e.g. `/monitoring#silences`.
+ */
+@Directive({
+  selector: '[cdTabActivateByFragment]'
```

Comment:
I think we could even 'automagically' apply this to all `<tabs>` by simply reusing the same selector pattern that 'TabDirective uses (Angular allows to apply multiple directives to the same selector):
 
```suggestion
  selector: 'tab:[tab]'
```

I was also going to suggest using inheritance here (instead of DI, so you don't need to redefine `@Input() fragment`), but given we don't control ngx-bootstrap, it think this is a safer approach.

Reply:
Will this also apply to tabs inside details? don't think we should do this.